presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DesugarAtTimeZone.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.DesugaringRewriter;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
+import com.facebook.presto.sql.tree.GroupingOperation;
+import com.facebook.presto.sql.tree.NodeRef;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+import java.util.Map;
+
+import static com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+public class DesugarAtTimeZone
+        extends ExpressionRewriteRuleSet
+{
+    public DesugarAtTimeZone(Metadata metadata, SqlParser sqlParser)
+    {
+        super((expression, context) -> rewrite(expression, context, metadata, sqlParser));
+    }
+
+    static Expression rewrite(Expression expression, Rule.Context context, Metadata metadata, SqlParser sqlParser)
```

Comment:
I would prefer not to copy paste code from legacy rules. Do the same thing as in other review, extract a legacy rewrite as some class and then use it here.